auth/cache_test.go

```diff
@@ -13,189 +13,283 @@ import (
+				randomDataJSON, err := json.Marshal(randomData)
+				require.NoError(err)
 
-	cache2 := Cache{
-		AccessToken:       "AnotherTopSecret!",
-		RefreshToken:      "StillSoRefreshing:)",
-		AccessTokenExpiry: now,
-		SessionExpiry:     now,
+				err = os.MkdirAll(dirPath, directoryPermissions)
+				require.NoError(err)
+				err = os.WriteFile(credPath, randomDataJSON, directoryPermissions)
+				require.NoError(err)
+
+				return nil
+			},
+			expectedError: "bad format: access token is not a valid JWT",
+		},
 	}
-	assert.NoError(Write(cache2))
 
-	assert.DirExists(credentialDirectory)
-	assert.FileExists(credentialPath)
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
 
-	rawJSON, err := os.ReadFile(credentialPath)
-	require.NoError(err)
+			credDir, credPath, err := Setup(t)
+			require.NoError(err)
+			require.NotNil(credDir)
 
-	var cacheFromJSON Cache
+			// Runs specific test case setup.
+			err = testCase.caseSetup(credDir, credPath)
+			require.NoError(err)
 
-	err = json.Unmarshal(rawJSON, &cacheFromJSON)
-	require.NoError(err)
+			// Run the test.
+			_, err = Read()
 
-	assert.Equal(cache2.AccessToken, cacheFromJSON.AccessToken)
-	assert.Equal(cache2.RefreshToken, cacheFromJSON.RefreshToken)
-	assert.Equal(cache2.AccessTokenExpiry.Format("2006-01-02T15:04:05 -07:00:00"), cacheFromJSON.AccessTokenExpiry.Format("2006-01-02T15:04:05 -07:00:00"))
-	assert.Equal(cache2.SessionExpiry.Format("2006-01-02T15:04:05 -07:00:00"), cacheFromJSON.SessionExpiry.Format("2006-01-02T15:04:05 -07:00:00"))
-	require.NoError(destroy())
-}
+			// Make assertions
+			assert.Error(err)
+			assert.ErrorContains(err, testCase.expectedError)
 
-func TestWrite_NoDirectoryNoFile(t *testing.T) {
-	require := requirepkg.New(t)
-	assert := assertpkg.New(t)
+			// Clean up.
+			require.NoError(Destroy(t))
 
-	credentialDirectory, credentialPath, err := setup()
-	require.NoError(err)
-	require.NotNil(credentialDirectory)
+		})
 
-	now := time.Now()
-	cache := Cache{
-		AccessToken:       "TopSecret!",
-		RefreshToken:      "SoRefreshing:)",
-		AccessTokenExpiry: now,
-		SessionExpiry:     now,
 	}
 
-	assert.NoError(Write(cache))
-	assert.DirExists(credentialDirectory)
-	assert.FileExists(credentialPath)
-
-	rawJSON, err := os.ReadFile(credentialPath)
-	require.NoError(err)
-
-	var cacheFromJSON Cache
-
-	err = json.Unmarshal(rawJSON, &cacheFromJSON)
-	require.NoError(err)
-
-	assert.Equal(cache.AccessToken, cacheFromJSON.AccessToken)
-	assert.Equal(cache.RefreshToken, cacheFromJSON.RefreshToken)
-	assert.Equal(cache.AccessTokenExpiry.Format("2006-01-02T15:04:05 -07:00:00"), cacheFromJSON.AccessTokenExpiry.Format("2006-01-02T15:04:05 -07:00:00"))
-	assert.Equal(cache.SessionExpiry.Format("2006-01-02T15:04:05 -07:00:00"), cacheFromJSON.SessionExpiry.Format("2006-01-02T15:04:05 -07:00:00"))
-	require.NoError(destroy())
 }
 
-func TestWrite_DirectoryExistsNoFile(t *testing.T) {
+func TestRead_ValidFormat(t *testing.T) {
```

Comment:
In keeping with the pattern in the rest of the file, let's place the valid format test ahead of the invalid format tests, so someone reading through the test first learns correct format first.